diff --git a/gpio-ch341.c b/gpio-ch341.c
index de0184a..5b332b1 100644
--- a/gpio-ch341.c
+++ b/gpio-ch341.c
@@ -212,7 +212,7 @@ static void write_outputs(struct ch341_gpio *dev)
 	mutex_unlock(&dev->gpio_lock);
 }
 
-static void ch341_gpio_set(struct gpio_chip *chip, unsigned int offset, int value)
+static int ch341_gpio_set(struct gpio_chip *chip, unsigned int offset, int value)
 {
 	struct ch341_gpio *dev = gpiochip_get_data(chip);
 
@@ -222,9 +222,10 @@ static void ch341_gpio_set(struct gpio_chip *chip, unsigned int offset, int valu
 		dev->gpio_last_written &= ~BIT(offset);
 
 	write_outputs(dev);
+  return 0;
 }
 
-static void ch341_gpio_set_multiple(struct gpio_chip *chip,
+static int ch341_gpio_set_multiple(struct gpio_chip *chip,
 				    unsigned long *mask, unsigned long *bits)
 {
 	struct ch341_gpio *dev = gpiochip_get_data(chip);
@@ -232,6 +233,7 @@ static void ch341_gpio_set_multiple(struct gpio_chip *chip,
 	dev->gpio_last_written = (dev->gpio_last_written & ~*mask) | (*bits & *mask);
 
 	write_outputs(dev);
+  return 0;
 }
 
 static int ch341_gpio_get_direction(struct gpio_chip *chip,
