diff --git a/amifldrv.c b/amifldrv.c
index 394e35a..45e1556 100644
--- a/amifldrv.c
+++ b/amifldrv.c
@@ -354,7 +354,11 @@ int AMI_chrdrv_mmap(struct file *file, struct vm_area_struct *vma)
 		return (-EINVAL);
 	}
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 3, 0)
+	vm_flags_set(vma, VM_LOCKED);
+#else
 	vma->vm_flags |= VM_LOCKED;
+#endif
 
 	printk(KERN_INFO "%s:%d Kmalloc area %p\n",
 			__FUNCTION__, __LINE__, kmalloc_area);

