diff --git a/ipa-httpd.cnf b/ipa-httpd.cnf
index 96c1bed..7ad9752 100644
--- a/ipa-httpd.cnf
+++ b/ipa-httpd.cnf
@@ -1,5 +1,5 @@
 # the fully qualified server (or service) name
-FQDN = server.example.test
+FQDN = ipa.harkema.io
 ALTNAMES = DNS:$FQDN
 
 # --- no modifications required below ---
diff --git a/renew-le.sh b/renew-le.sh
index 9783877..0ce91b6 100755
--- a/renew-le.sh
+++ b/renew-le.sh
@@ -2,7 +2,7 @@
 set -o nounset -o errexit
 
 WORKDIR=$(dirname "$(realpath $0)")
-EMAIL=""
+EMAIL="tomas@harkema.io"
 
 ### cron
 # skip renewal if the cert is still valid for more than 30 days
@@ -36,7 +36,7 @@ else
 fi
 
 # get a new cert
-letsencrypt certonly --standalone --csr "$WORKDIR/httpd-csr.der" --email "$EMAIL" --agree-tos
+letsencrypt certonly --manual --preferred-challenges dns --csr "$WORKDIR/httpd-csr.der" --email "$EMAIL" --agree-tos
 
 # replace the cert
 cp /var/lib/ipa/certs/httpd.crt /var/lib/ipa/certs/httpd.crt.bkp
diff --git a/setup-le.sh b/setup-le.sh
index ea2a694..6f78cfd 100755
--- a/setup-le.sh
+++ b/setup-le.sh
@@ -6,7 +6,7 @@ WORKDIR=$(dirname "$(realpath $0)")
 CERTS=("isrgrootx1.pem" "isrg-root-x2.pem" "lets-encrypt-r3.pem" "lets-encrypt-e1.pem" "lets-encrypt-r4.pem" "lets-encrypt-e2.pem")
 
 sed -i "s/server.example.test/$FQDN/g" $WORKDIR/ipa-httpd.cnf
-
+dnf install epel-release -y
 dnf install letsencrypt -y
 
 if [ ! -d "/etc/ssl/$FQDN" ]
