diff --git a/configure.ac b/configure.ac
index 57a82b7..13b8180 100644
--- a/configure.ac
+++ b/configure.ac
@@ -173,6 +173,16 @@ AC_ARG_WITH([dbusinterfacedir],
         [], [with_dbusinterfacedir=${datadir}/dbus-1/interfaces])
 AC_SUBST([dbusinterfacedir], [$with_dbusinterfacedir])
 
+AC_ARG_WITH([dbusabrtconfdir],
+        AS_HELP_STRING([--with-dbusabrtconfdir=DIR], [Directory for dbus interface files]),
+        [], [with_dbusabrtconfdir=${sysconfdir}/dbus-1/interfaces])
+AC_SUBST([dbusabrtconfdir], [$with_dbusabrtconfdir])
+
+AC_ARG_WITH([autostartdir],
+        AS_HELP_STRING([--with-autostartdir=DIR], [autostartdir]),
+        [], [with_autostartdir=${sysconfdir}/xdg/autostart])
+AC_SUBST([autostartdir], [$with_autostartdir])
+
 AC_ARG_WITH(largedatatmpdir,
             [AS_HELP_STRING([--with-largedatatmpdir=DIR],
                            [Directory where potentially large data are created (default: /var/tmp)])],
diff --git a/src/daemon/Makefile.am b/src/daemon/Makefile.am
index c39333f..b9eaf32 100644
--- a/src/daemon/Makefile.am
+++ b/src/daemon/Makefile.am
@@ -136,7 +136,7 @@ dist_daemonconf_DATA = \
     abrt-action-save-package-data.conf \
     gpg_keys.conf
 
-dbusabrtconfdir = ${sysconfdir}/dbus-1/system.d/
+# dbusabrtconfdir = ${dbusabrtconfdir}
 dist_dbusabrtconf_DATA = org.freedesktop.problems.daemon.conf
 
 EXTRA_DIST = abrt-handle-upload.in
diff --git a/src/dbus/Makefile.am b/src/dbus/Makefile.am
index fe71dcb..1cfba95 100644
--- a/src/dbus/Makefile.am
+++ b/src/dbus/Makefile.am
@@ -58,7 +58,7 @@ DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
 polkitconfdir = ${datadir}/polkit-1/actions
 dist_polkitconf_DATA = abrt_polkit.policy
 
-dbusabrtconfdir = ${sysconfdir}/dbus-1/system.d/
+# dbusabrtconfdir = ${dbusabrtconfdir}
 dist_dbusabrtconf_DATA = dbus-abrt.conf
 
 orgfreedesktopproblemsservicedir = ${datadir}/dbus-1/system-services
